### ISSUE TITLE
A plugin to monitor websites' response time, health etc. Something close to what traceroute does. 

### ISSUE THREAD

**Issue by abhi-jha** [influxdata/telegraf#12921]
### Use Case

Curently we can do this via the `exec` plugin as described below.

```
[[inputs.exec]]
      interval = "60s"
      commands=["sudo mtr -C -n google.com"]
      timeout = "40s"
      data_format = "csv"
      csv_skip_rows = 1
      csv_column_names=["", "", "status", "dest", "hop", "ip", "loss", "snt", "", "", "avg", "best", "worst", "stdev"]
      name_override = "mtr"
      csv_tag_columns = ["dest", "hop", "ip"]
```

But this requires telegraf being having `sudo` user access so that it can execute `mtr` command. 

The idea is to get this functionality in a native plugin. 

Example new plugin : 

[[inputs.trace]]
      interval = "60s"
      sites=["google.com", "github.com", "reddit.com"]
      timeout = "40s" # timeout per site can be done using this plugin multiple times for each site requiring a different timeout.
      column_names=["", "", "status", "dest", "hop", "ip", "loss", "snt", "", "", "avg", "best", "worst", "stdev"]
      
``` 

### Expected behavior

Not sure but it could look like how `mtr` provides output: 

`trace,dest=google.com,hop=7,host=hostname.local,ip=2a00:1450:8106::1 avg=19.83,best=14.22,worst=37.31,stdev=6.73,status="OK",loss=10,snt=10i 1679419996000000000`


It maybe possible to include how many hops it is taking as well, something that `dig` does when doing recurisve search to root servers.

### Actual behavior

I used `mtr` within `exec` for now but it required that I make the current user which ran telegraf a sudoer. 

Making it a native plugin that is only dependent on libraries eliminates the need to have it within `sudoer`

### Additional info

```
mtr,dest=facebook.com,hop=10,host=host.local,ip=2a03:2880:f173:81:face:b00c:0:25de status="OK",loss=0,snt=10i,avg=15.05,best=13.42,worst=17.87,stdev=1.34 1679419996000000000
mtr,dest=google.com,hop=1,host=host.local,ip=2a02:a210:abd:2600:aef8:ccff:feeb:c984 avg=4.86,best=4.21,worst=5.51,stdev=0.92,status="OK",loss=80,snt=10i 1679419996000000000
mtr,dest=reddit.com,hop=2,host=host.local,ip=??? stdev=0,status="OK",loss=100,snt=10i,avg=0,best=0,worst=0 1679420055000000000
```

**Comment by powersj** [comment]
> A plugin to monitor websites' response time, health etc. Something close to what traceroute does.

There are like 3 different requests in that title :) What are you actually trying to do or what is your end goal metric? And have you looked at the ping plugin?

**Comment by Hipska** [comment]
On slack we only discussed for a plugin that can do trace routes natively instead of via the exec command.

Response time and health can already be done via existing plugins.

**Comment by abhi-jha** [comment]
Yeah, lets reduce the scope for this. I am fine with doing just the traceroute thing.

**Comment by abhi-jha** [comment]
@powersj Let me know if we need to discuss this more. I am looking around for some libraries that can provide  the `traceroute` metrics. Do let me know if you have some ideas to pick a particular one. 

I can start with a PR.

**Comment by powersj** [comment]
> Let me know if we need to discuss this more. I am looking around for some libraries that can provide the traceroute metrics

I would like to see what you find, especially if it does not require sudo.

You can see https://github.com/traviscross/mtr/issues/204#issuecomment-723961118 about why mtr requires sudo and the comments about setting setuid or other privliges can remove the need for sudo. Again, a user could use this method, today.

**Comment by abhi-jha** [comment]
Should I go ahead and give pro-bing a shot? I want to see what the exact would look like.

**Comment by Hipska** [comment]
Sure, let us know if it doesn’t work out, we can look for alternatives or solutions.

**Comment by abhi-jha** [comment]
Is there a suggested list of metrics that comes to mind as ouptut? 

`count of hops, time taken(min, max, avg, sd), result_code,packets_sent, packets_received, ttl`

Is this the right direction?

**Comment by Hipska** [comment]
If it could be like you showed as example in “Additional info”, that would be great!

**Comment by srebhan** [comment]
Note to myself: https://github.com/wisdomatom/go-mtr

**Comment by abhi-jha** [comment]
Hello. Looks like I missed this for some time. Is there already such a plugin in? Or should I continue ahead?

**Comment by Hipska** [comment]
There is no mtr or trace route related plugin yet, You are free to go ahead 😉

**Comment by abhi-jha** [comment]
Hello. I am trying to get pro-bing library to work as a small POC.
